### PR TITLE
(Planner 4): Secure planner model is recalculated and text is changed

### PR DESCRIPTION
**Pull request by janiversen**
### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [x ] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

When changing METERS <-> FEET, secure that the model is recalculated and that the 
UI text is changed.

### CODE REVIEW

File: backend-shared/plannershared.cpp [hunk 1]
```diff
@@ -301,3 +309,9 @@ void plannerShared::set_display_variations(bool value)
 {
 	DivePlannerPointsModel::instance()->setDisplayVariations(value);
 }
+
+// Handle when user change length measurement type
+void plannerShared::unit_lengthChangedSlot(int value)
+{
+	qDebug() << "---> change METER <-> FEET";
```
Comment (dirkhh) [inline backend-shared/plannershared.cpp]: I don't think we want to check this into production code...

File: backend-shared/plannershared.cpp [hunk 2]
```diff
@@ -313,5 +313,10 @@ void plannerShared::set_display_variations(bool value)
 // Handle when user change length measurement type
 void plannerShared::unit_lengthChangedSlot(int value)
 {
-	qDebug() << "---> change METER <-> FEET";
+	// Provoke recalculation of model and send of signals
+	set_ascratelast6m(ascratelast6m());
+	set_ascratestops(ascratestops());
+	set_ascrate50(ascrate50());
+	set_ascrate75(ascrate75());
+	set_descrate(descrate());
```
Comment (dirkhh) [inline backend-shared/plannershared.cpp:320]: please collapse this with the previous commit and remove the `#include <QDebug>`

File: mobile-widgets/qml/DivePlannerSetup.qml [hunk 3]
```diff
@@ -11,11 +11,19 @@ Kirigami.ScrollablePage {
...
 		spinAscrate75.value--
+		spinAscrate50.value++
+		spinAscrate50.value--
+		spinAscratestops.value++
+		spinAscratestops.value--
+		spinAscratelast6m.value++
+		spinAscratelast6m.value--
+		spinDescrate.value++
+		spinDescrate.value--
+		console.log("---> new values READY")
```
Comment (dirkhh) [inline mobile-widgets/qml/DivePlannerSetup.qml]: again, this doesn't look like the final code. If the increment/decrement pairs are needed, then we should have comments there explaining this

**Review by dirkhh — CHANGES_REQUESTED** [review]
two of the commits should be combined... and more questions

File: backend-shared/plannershared.cpp (hunk 1 above)
Comment (janiversen) [reply backend-shared/plannershared.cpp]: It was just there to prove that the signal got received, I had some serious problems with qPrefUnit. 

Gone

File: backend-shared/plannershared.cpp (hunk 2 above)
Comment (janiversen) [reply backend-shared/plannershared.cpp:320]: done

File: mobile-widgets/qml/DivePlannerSetup.qml (hunk 3 above)
Comment (janiversen) [reply mobile-widgets/qml/DivePlannerSetup.qml]: Added a longer comment in one of the previous commits. Apart from the console.log (which is gone) this is the final code. At least until I find a better way to make the spinBox update its text.

**Comment by janiversen** [comment]
I should have replied to all comment in the 4 PRs, and in most cases done the requested changes.

At some point we need to make a design decision on how to handle C based enums in QML. QML requires the enum is defined in a class with an Q_OBJECT macro. This is pretty easy to do, and the C code can still access it (because we compile it as C++ code), but it is a change that does a search/replace in a LOT of files.

The other alternative (as used in qPrefCloudStorage) is to duplicate the enum, and add a big warning. That solution is pretty easy but also dangerous (to a degree, because how often do we change these fundamental enums). I have done that in my alternative implementation, and are happy to submit that code as a new PR (which includes fixing e.g. qPrefUnit).

**Comment by dirkhh** [comment]
I did some small rework of the whole series and pushed it to master.

**Pull request closed without merging**